### DISCUSSION TITLE
How to directly connect to a bonded (server) device as client

### DISCUSSION THREAD

**Discussion by romwam** [h2zero/NimBLE-Arduino#716] [Q&A]
I have a client (ESP32S3) which is already bonded to a BLE server device (Bonding is stored in NVS, bonding information is also properly restored upon start of NimBLE stack). I also know the BLE server device's address upon start (by persisting this address in my preference settings)

What is the recommended way to connect to this server device using the known address and available bonding without having to start the BLE scan procedure?

**Comment by h2zero** [comment]
You can do it this way: `pClient->connect(addr);`. https://h2zero.github.io/NimBLE-Arduino/class_nim_b_l_e_client.html#a026d7e04557b4821a7da2c1a1caa46eb
